test/lib/PriorityFeeLib.t.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {console2} from "forge-std/console2.sol";
+import {Test} from "forge-std/Test.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {InputToken, OutputToken} from "../../src/base/ReactorStructs.sol";
+import {PriorityInput, PriorityOutput} from "../../src/lib/PriorityOrderLib.sol";
+import {PriorityFeeLib} from "../../src/lib/PriorityFeeLib.sol";
+
+// base fee is 0 by default
+contract PriorityFeeLibTest is Test {
+    uint256 constant MPS = 1e7;
+
+    function testScaleInputNoPriorityFee() public {
+        assertEq(tx.gasprice, 0);
+
+        PriorityInput memory input =
+            PriorityInput({token: ERC20(address(0)), amount: 1 ether, mpsPerPriorityFeeWei: 100});
+
+        uint256 scaledAmount = (input.amount * (MPS - tx.gasprice * input.mpsPerPriorityFeeWei)) / MPS;
+
+        InputToken memory scaledInput = PriorityFeeLib.scale(input, tx.gasprice);
+        assertEq(scaledInput.amount, scaledAmount);
+        assertEq(scaledInput.maxAmount, input.amount);
+    }
+
+    function testScaleOutputNoPriorityFee() public {
+        assertEq(tx.gasprice, 0);
+
+        PriorityOutput memory output =
+            PriorityOutput({token: address(0), amount: 1 ether, mpsPerPriorityFeeWei: 100, recipient: address(0)});
+
+        OutputToken memory scaledOutput = PriorityFeeLib.scale(output, tx.gasprice);
+        uint256 scaledAmount = (output.amount * (MPS + tx.gasprice * output.mpsPerPriorityFeeWei)) / MPS;
+
+        assertEq(scaledOutput.amount, scaledAmount);
+    }
+
+    function testScaleInputPriorityFee_fuzz(uint256 priorityFee) public {
```

Comment:
some non-fuzz edge case tests would be good:
- input fee over max defaults to 0
- super high / low priority fee

Reply:
I added explicit tests for when the mpsPerPriorityFeeWei is 0, when the amount is large enough to trigger overflow in the math library, for small priority fees, and for when input fee defaults to 0